src/main/java/net/snowflake/client/core/JsonSqlInput.java

```diff
@@ -0,0 +1,341 @@
+    return withNextValue(
+        (value, jsonNode, fieldMetadata) -> {
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          return mapExceptions(() -> converters.getNumberConverter().getShort(value, columnType));
+        });
+  }
+
+  @Override
+  public int readInt() throws SQLException {
+    return withNextValue(
+        (value, jsonNode, fieldMetadata) -> {
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          return mapExceptions(() -> converters.getNumberConverter().getInt(value, columnType));
+        });
+  }
+
+  @Override
+  public long readLong() throws SQLException {
+    return withNextValue(
+        (value, jsonNode, fieldMetadata) -> {
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          return mapExceptions(() -> converters.getNumberConverter().getLong(value, columnType));
+        });
+  }
+
+  @Override
+  public float readFloat() throws SQLException {
+    return withNextValue(
+        (value, jsonNode, fieldMetadata) -> {
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          return mapExceptions(() -> converters.getNumberConverter().getFloat(value, columnType));
+        });
+  }
+
+  @Override
+  public double readDouble() throws SQLException {
+    return withNextValue(
+        (value, jsonNode, fieldMetadata) -> {
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          return mapExceptions(() -> converters.getNumberConverter().getDouble(value, columnType));
+        });
+  }
+
+  @Override
+  public BigDecimal readBigDecimal() throws SQLException {
+    return withNextValue(
+        (value, jsonNode, fieldMetadata) -> {
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          return mapExceptions(
+              () -> converters.getNumberConverter().getBigDecimal(value, columnType));
+        });
+  }
+
+  @Override
+  public byte[] readBytes() throws SQLException {
+    return withNextValue(
+        (value, jsonNode, fieldMetadata) -> {
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          int columnSubType = fieldMetadata.getType();
+          int scale = fieldMetadata.getScale();
+          return mapExceptions(
+              () ->
+                  converters.getBytesConverter().getBytes(value, columnType, columnSubType, scale));
+        });
+  }
+
+  @Override
+  public Date readDate() throws SQLException {
+    return withNextValue(
+        (value, jsonNode, fieldMetadata) -> {
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          int columnSubType = fieldMetadata.getType();
+          TimeZone tz = TimeZone.getDefault(); // TODO structuredType how to get timezone?
```

Comment:
when/in which issue this TODO will be implemented?

Reply:
SNOW-974575

Reply:
I removed unused fields and created jira SNOW-1055577

Reply:
please take a look at other TODO comments - should they be there? can we add issue to them?

Reply:
Removed. It will be important when timezone won't be sent as String